src/java/org/apache/cassandra/service/accord/AccordService.java

```diff
@@ -973,6 +1096,30 @@ private static CommandStoreTxnBlockedGraph.TxnState populate(CommandStoreTxnBloc
         return cmdTxnState.build();
     }
 
+    @Nullable
+    @Override
+    public Long minEpoch(Collection<TokenRange> ranges)
+    {
+        return node.topology().minEpoch();
```

Comment:
Benedict said not to be fancy and just return the min known.... we do *not* evict epochs atm, so this is old in long running clusters, but the argument is that we will be able to purge once durability scheduling is working... we need the older epochs to be `redundant` (aka nothing is ever going to touch them) so we can purge...

TL;DR - atm this is returning very old epochs, but once other things come into play this will return more "recent" epochs.  That something is out of scope for this work